### PR TITLE
breezy 3.3.0

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
same `pyproject.toml` build issue
```
    × Building wheel for breezy (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> See above for output.
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
    full command: /home/linuxbrew/.linuxbrew/Cellar/breezy/3.3.0/libexec/bin/python3.10 /home/linuxbrew/.linuxbrew/Cellar/breezy/3.3.0/libexec/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpzd9k0w5v
    cwd: /tmp/breezy-20221009-10261-1j1kp86/breezy-3.3.0
    Building wheel for breezy (pyproject.toml): finished with status 'error'
```

**Comment by bayandin** [comment]
```
  ==> Testing breezy
  ==> /opt/homebrew/Cellar/breezy/3.3.0/bin/brz whoami Homebrew <homebrew@example.com>
  brz: ERROR: Couldn't import breezy and dependencies.
  Please check the directory containing breezy is on your PYTHONPATH.
  
  Error: PyErr { type: <class 'ModuleNotFoundError'>, value: ModuleNotFoundError("No module named 'breezy'"), traceback: None }
  Error: breezy: failed
```

`bin/brz` is not a python script anymore, it's a binary which gets linked with python, but python is not from venv, and this causes the test to fail.

```
% otool -L bin/brz
bin/brz:
	/opt/homebrew/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/Python (compatibility version 3.10.0, current version 3.10.0)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/usr/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 1.0.0)
```

**Comment by cho-m** [comment]
As `virtualenv_install_with_resources` creates symlinks in `bin`, it should be possible to replace the symlinks with env scripts that set either `PYTHONPATH=libexec/site_packages` or `PATH=libexec/bin` to help binary pick up the virtualenv.

File: Formula/breezy.rb [hunk 1]
```diff
@@ -20,45 +20,50 @@ class Breezy < Formula
   end
 
   depends_on "gettext" => :build
-  depends_on "libcython" => :build
```
Comment (cho-m) [inline Formula/breezy.rb:23]: I removed this as `libcython` is keg-only so wasn't used during build.

**Review by branchv — APPROVED** [review]

**Review by branchv — APPROVED** [review]

File: Formula/breezy.rb [hunk 2]
```diff
@@ -20,45 +20,50 @@ class Breezy < Formula
...
+    url "https://files.pythonhosted.org/packages/b2/56/d87d6d3c4121c0bcec116919350ca05dc3afd2eeb7dc88d07e8083f8ea94/urllib3-1.26.12.tar.gz"
+    sha256 "3fa96cf423e6987997fc326ae8df396db2a8b7c667747d47ddd8ecba91f4a74e"
   end
 
   def install
     virtualenv_install_with_resources
+    bin.each_child do |f|
+      f.unlink
+      f.write_env_script libexec/"bin"/f.basename, PATH: "#{libexec}/bin:$PATH"
+    end
```
Comment (cho-m) [inline Formula/breezy.rb:66]: Taking another look at binaries, we could just restrict this to only `brz` as others are still python scripts
```console
❯ ls bin
brz  bzr  bzr-receive-pack  bzr-upload-pack  git-remote-bzr

❯ file libexec/bin/*bzr*
libexec/bin/bzr-receive-pack: a @@HOMEBREW_CELLAR@@/breezy/3.3.0/libexec/bin/python3.11 script text executable, ASCII text
libexec/bin/bzr-upload-pack:  a @@HOMEBREW_CELLAR@@/breezy/3.3.0/libexec/bin/python3.11 script text executable, ASCII text
libexec/bin/git-remote-bzr:   a @@HOMEBREW_CELLAR@@/breezy/3.3.0/libexec/bin/python3.11 script text executable, ASCII text

❯ file libexec/bin/*brz*
libexec/bin/brz: Mach-O 64-bit executable arm64
```

So an alternative could be:
```suggestion
    (bin/"brz").unlink
    (bin/"brz").write_env_script libexec/"bin/brz", PATH: "#{libexec}/bin:$PATH"
```

Comment (cho-m) [reply Formula/breezy.rb:66]: Though it might be safer to keep current logic in case any Python script has code that shells out to `brz`.

**Pull request closed without merging**